### ISSUE TITLE
org.openrewrite.java.spring.boot3.UpgradeSpringBoot_3_2 changes build.gradle sourceCompatibility and targetCompatibility from 21 back to 17

### ISSUE THREAD

**Issue by aholland** [openrewrite/rewrite-spring#477]
Title says it all - sorry for the brief report! Also I think they should ideally be JavaVersion.21 in Gradle 8.5 (to be fair you might already be detecting which convention to use - I had a plain `21` before I ran the script).

**Comment by timtebeek** [comment]
You're right, that's a bug; pending a fix & release you might want to run this recipe instead, as that uses Java 21
https://docs.openrewrite.org/recipes/java/spring/boot3/springboot3bestpractices
